### PR TITLE
requests: annotate RequestsCookieJar

### PR DESCRIPTION
**Pull request by principis**
Adds type annotations for RequestsCookieJar.

At first one would think RequestsCookieJar supports `str | bytes` as cookie value, but this is not the case. There is a `startswith` check in  [set_cookie](https://github.com/psf/requests/blob/v2.31.0/requests/cookies.py#L352) that prevents using `bytes`.

### CODE REVIEW

File: stubs/requests/requests/cookies.pyi [hunk 1]
```diff
@@ -33,26 +35,26 @@ def remove_cookie_by_name(cookiejar, name, domain=None, path=None): ...
...
+    def values(self) -> list[str]: ...  # type: ignore[override]
+    def iteritems(self) -> Iterator[tuple[str, str]]: ...
+    def items(self) -> list[tuple[str, str]]: ...  # type: ignore[override]
+    def list_domains(self) -> list[str]: ...
+    def list_paths(self) -> list[str]: ...
+    def multiple_domains(self) -> bool: ...
+    def get_dict(self, domain=None, path=None) -> dict[str, str]: ...
+    def __getitem__(self, name) -> str: ...
     def __setitem__(self, name, value) -> None: ...
     def __delitem__(self, name) -> None: ...
```
Comment (hauntsaninja) [inline stubs/requests/requests/cookies.pyi:53]: want to add arg annotations to these ones as well?

**Review by hauntsaninja — COMMENTED** [review]
Thanks for the improvement!

File: stubs/requests/requests/cookies.pyi (hunk 1 above)
Comment (principis) [reply stubs/requests/requests/cookies.pyi:53]: I forgot those, thanks!

**Comment by principis** [comment]
Added the requested changes and rebased the branch.

**Review by srittau — APPROVED** [review]
Thanks!

**Pull request merged**